cf-java-logging-support-logback/src/main/java/com/sap/hcp/cf/logback/encoder/BaseFieldSupplier.java

```diff
@@ -0,0 +1,40 @@
+package com.sap.hcp.cf.logback.encoder;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.sap.hcp.cf.logback.converter.api.LogbackContextFieldSupplier;
+import com.sap.hcp.cf.logging.common.Defaults;
+import com.sap.hcp.cf.logging.common.Fields;
+import com.sap.hcp.cf.logging.common.Markers;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+
+public class BaseFieldSupplier implements LogbackContextFieldSupplier {
+
+    @Override
+    public Map<String, Object> map(ILoggingEvent event) {
+        Map<String, Object> fields = new HashMap<>(6);
+        fields.put(Fields.WRITTEN_AT, Instant.ofEpochMilli(event.getTimeStamp()).toString());
+        fields.put(Fields.WRITTEN_TS, now());
+        fields.put(Fields.TYPE, isRequestLog(event) ? Defaults.TYPE_REQUEST : Defaults.TYPE_LOG);
+        fields.put(Fields.LEVEL, String.valueOf(event.getLevel()));
+        fields.put(Fields.LOGGER, event.getLoggerName());
+        if (!isRequestLog(event)) {
+            fields.put(Fields.MSG, event.getFormattedMessage());
+        }
+        return fields;
+    }
+
+    private String now() {
+        Instant now = Instant.now();
+        long timestamp = now.getEpochSecond() * 1_000_000_000L + now.getNano();
```

Comment:
The Java now implementation did not have different rollover times right?
So there will be no issue with get Seconds rolling over differently then Nano, resulting in false reported order of logs.

Reply:
The Javadoc of the `Instant` methods makes it pretty clear, that `getNano()` will only return values between 0 and 999,999,999. There are no problems with separate roll-overs here.